tests/unit/app-template_test.cc

```diff
@@ -24,22 +24,21 @@
 #include <string>
 #include <boost/test/unit_test.hpp>
 #include <seastar/core/app-template.hh>
+#include <seastar/core/sleep.hh>
 
 using namespace seastar;
+using namespace std::chrono_literals;
 
-#ifdef Seastar_DPDK
-constexpr bool with_dpdk = true;
-#else
-constexpr bool with_dpdk = false;
-#endif
-
-BOOST_AUTO_TEST_CASE(app_standard_memory_allocator,
-                     *boost::unit_test::enable_if<with_dpdk>()) {
+// #2148 - always run this.
+BOOST_AUTO_TEST_CASE(app_standard_memory_allocator) {
     // by default, use conservative settings instead of maxing out the performance
     // for testing app_template and underlying reactor's handling of different
     // settings
     app_template::seastar_options opts;
-    opts.smp_opts.thread_affinity.set_value(false);
+    // Here was a `opts.smp_opts.thread_affinity.set_value(false);`.
+    // This breaks the subsequent tests below (which run non-smp-1 config).
```

Comment:
i am not following you. yes, we do have non-smp-1 tests in this source file. but why does your change break them?

Reply:
It does not. But running the tests with and without it does. Something gets confused with some thread-local somewhere, such that the latter tests cannot init, because smp::invoke_all fails to complete.

Reply:
In fact, the service_group semaphores gets confused.

Reply:
I changed my mind. Better to fix the issue.

Reply:
thanks!